test/dreyfus_blacklist_request_test.erl

```diff
@@ -56,18 +56,16 @@ deny_fabric_requests() ->
     Reason = <<"Index <mydb, myddocid, myindexname>, is BlackListed">>,
     Index = ["mydb", "myddocid", "myindexname"],
     QueryArgs = #index_query_args{},
-    dreyfus_test_util:with_config_listener(fun() ->
-        config:set("dreyfus", "black_list", [Index]),
-        dreyfus_test_util:wait_for_config(),
-        ?assertThrow({bad_request, Reason}, dreyfus_fabric_search:go(<<"mydb">>,
-            <<"myddocid">>,  <<"myindexname">>, QueryArgs)),
-        ?assertThrow({bad_request, Reason}, dreyfus_fabric_group1:go(<<"mydb">>,
-            <<"myddocid">>,  <<"myindexname">>, QueryArgs)),
-        ?assertThrow({bad_request, Reason}, dreyfus_fabric_group2:go(<<"mydb">>,
-            <<"myddocid">>,  <<"myindexname">>, QueryArgs)),
-        ?assertThrow({bad_request, Reason}, dreyfus_fabric_info:go(<<"mydb">>,
-            <<"myddocid">>,  <<"myindexname">>, QueryArgs))
-    end).
+    config:set("dreyfus_blacklist", Index, "true"),
+    dreyfus_test_util:wait_config_change(Index, "true"),
+    ?assertThrow({bad_request, Reason}, dreyfus_fabric_search:go(<<"mydb">>,
```

Comment:
I am having the following failure:
```
    dreyfus_blacklist_request_test: dreyfus_blacklist_request_test_...*failed*
in function dreyfus_blacklist_request_test:'-deny_fabric_requests/0-fun-0-'/0 (test/dreyfus_blacklist_request_test.erl, line 61)
in call from dreyfus_blacklist_request_test:deny_fabric_requests/0 (test/dreyfus_blacklist_request_test.erl, line 61)
**error:{assertException,
    [{module,dreyfus_blacklist_request_test},
     {line,61},
     {expression,
         "dreyfus_fabric_search : go ( << \"mydb\" >> , << \"myddocid\" >> , << \"myindexname\" >> , QueryArgs )"},
     {pattern,"{ throw , { bad_request , Reason } , [...] }"},
     {unexpected_exception,
         {error,database_does_not_exist,
             [{mem3_shards,load_shards_from_db,"mydb",[{file,...},{...}]},
              {mem3_shards,load_shards_from_disk,1,[{...}|...]},
              {mem3_shards,for_db,2,[...]},
              {dreyfus_fabric_search,go,4,...},
              {dreyfus_blacklist_request_test,
                  '-deny_fabric_requests/0-fun-0-',...},
              {dreyfus_blacklist_request_test,...},
              {...}|...]}}]}
  output:<<"">>
```

It seems there are two problems here:
1. the `dreyfus_util:maybe_deny_index/4` is not called when `dreyfus_fabric_search:go/4` receives `#index_query_args{}`
2. The mock is not sufficient.